Normalize axis data

miscco opened the pull request:
this is a very hacky try to make datetick compilable for m2t inspired by #905.

The idea is simple. Tacke the rather stupid matlab format and project it into a reasonable range tixz can work with. There is obviously a big drawback, as it is done for all the the axes.

If the xaxis is in datetick one does
```matlab
cleanfigure('transformDates', 'x')
```

egeerardyn commented:
This just needs a bit of documentation (in the header of `cleanfigure`) to show how it can be used.

Note also that you can get it to work on a single axes object: `cleanfigure(hAx, ...)` should work without a problem. That is something you could also state in the docs.

miscco commented:
Yeah, thats true.

However, i thought about epanding it. In general, this has nothing to do with dateticks, but with a normalization of the axis data. So i am inklined to rename it to normalize_axis.

Also i believe i have to change the setup, so that it affects anything that has [upper(axis), 'Data'] in it. You know the figure setup better than me. Would it be sufficient to just go through all the direct children of the axis, or do i have to consider the grandchilds too?

egeerardyn commented:
@miscco I suppose that you need to handle all ancestors (but I imagine there are quite a lot of edge cases that this cannot handle: e.g. imagine some `hgtransform` somewhere, then you can no longer determine which dimension is affected).

Anyhow, since this is a work-around (and quite experimental), I'm fine with such limitations. Perhaps we should, however, add a warning to the documentation and/or at runtime to warn the users that this is experimental functionality.

miscco commented:
So i have reworked this to better express, what it is actually doing and added both the comments as well as the warning in the function itself. 

@egeerardyn I have no idea why the build failed. Can you enlighten me?

egeerardyn commented:
@miscco Jenkins configuration error. My bad (performed an update recently, apparently not the whole config was migrated). Will check out what is happening. Code itself ran fine, BTW.

egeerardyn commented:
@matlab2tikzBot: please test this

okomarov commented on src/cleanfigure.m:
```diff
@@ -22,6 +22,13 @@ function cleanfigure(varargin)
 %   or negative values disable this feature.
 %   (default: 1)
+%   CLEANFIGURE('normalizeAxis','xyz',...)
+%   EXPERIMENTAL: Normalize the data of the dimensions spezified by
```
typo: specified

egeerardyn commented:
@matlab2tikzBot: please test this

egeerardyn commented:
@miscco I updated Jenkins, this fixed the internal errors which caused the build to fail. Since I think Jenkins is a beast, I will look for alternatives that integrate better with GitHub when I find some time. A possible candidate would be TeamCity.

miscco commented:
So I found a  bug regarding the TickLabels. Not setting the TickLabelMode to manual changed the TickLabels to their new position. 

I tried a variety of ACID test that seemed to work fine. At least I did not find any corruption in MATLAB.

miscco commented:
Given this is experimental and we warn about it i will merge this tomorrow if noone objects

egeerardyn commented on src/cleanfigure.m:
```diff
@@ -1251,3 +1264,51 @@ function removeNaNs(meta, handle)
+    % Normalizes data from a given axis into the interval [0, 1]
+
+    % Warn about normalizeAxis being experimental
+    warning('cleanfigure:normalizeAxis', ...
```
👍

egeerardyn commented on src/cleanfigure.m:
```diff
@@ -1251,3 +1264,51 @@ function removeNaNs(meta, handle)
+    bool = length(val) <= 3;
+    for i=1:length(val)
+        bool = bool && ...
+               (strcmp(val(i), 'x') || ...
```
`strcmp` -> `strcmpi` (I will take care of this)
